### ISSUE TITLE
ES6 target is ignored when using TypeScript 1.8

### ISSUE THREAD

**Issue by tlancina** [ivogabe/gulp-typescript#296]
**Expected output:** 
```typescript
export * from 'A';
export * from 'B';
```

**Actual output:**
```js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
__export(require('A'));
__export(require('B'));
```

**Your gulpfile:**

```javascript
var gulp = require('gulp'),
    tsc = require('gulp-typescript');

gulp.task('test', function(){
  gulp.src('test.ts')
    .pipe(tsc({
      target: 'es6' // es2015 doesn't work either
    }))
    .pipe(gulp.dest('.'));
});
```

**tsc:** `tsc --target es6 test.ts`

**Code**

*Include your TypeScript code, if necessary.*
```typescript
export * from 'A';
export * from 'B';
```

This happens on previous versions as well, when using `typescript: require('typescript')` if TS 1.8 is installed locally.

**Comment by tlancina** [comment]
Looks like 
```
if (tsSettings.module === undefined) {
    // Same bug in TS 1.4 as previous comment.
	tsSettings.module = ts.ModuleKind.None;
}
```
in [main.ts](https://github.com/ivogabe/gulp-typescript/blob/402a8ee3facfe1b55cd0eb1b59c17488dc5cae43/lib/main.ts#L197-L200) is setting module to 0, which then causes typescript to ignore the target:
 ```
function getEmitModuleKind(compilerOptions) {
        return typeof compilerOptions.module === "number" ?
            compilerOptions.module :
            getEmitScriptTarget(compilerOptions) === 2 /* ES6 */ ? 5 /* ES6 */ : 1 /* CommonJS */;
    }
```

**Comment by ivogabe** [comment]
Fixed in #297
